scalyr_agent/builtin_monitors/journald_monitor.py

```diff
@@ -562,12 +556,22 @@ def _process_entries(self):
                 )
 
     def format_msg(
-        self, metric_name, metric_value, extra_fields=None,
+        self, metric_name, metric_value, log_config, extra_fields=None,
     ):
         string_buffer = six.StringIO()
 
-        if self._format_version == 2:
-            string_buffer.write('%s "%s"' % (metric_name, metric_value))
+        detect_escaped_strings = False
+        emit_raw_details = False
+        if log_config:
+            detect_escaped_strings = log_config.get("detect_escaped_strings", False)
```

Comment:
Add comment here `If True, will attempt to see if the values are already escaped strings, and if so, not add in an additional escaping ourselves`.